subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -392,19 +393,16 @@ STATIC int build_msg_block_for_send(struct lwm2m_message *msg, uint16_t block_nu
 STATIC int prepare_msg_for_send(struct lwm2m_message *msg)
 {
 	int ret;
-	uint16_t len;
-	const uint8_t *payload;
-
 	/* save the big buffer for later use (splitting blocks) */
 	msg->body_encode_buffer = msg->cpkt;
 
 	/* set the default (small) buffer for sending blocks */
 	msg->cpkt.data = msg->msg_data;
 	msg->cpkt.offset = 0;
-	msg->cpkt.max_len = MAX_PACKET_SIZE;
+	msg->cpkt.max_len = sizeof(msg->msg_data);
```

Comment:
In case of a block-wise transfer this should be set to `CONFIG_LWM2M_COAP_BLOCK_SIZE`. With your change `sizeof(msg->msg_data)` will always be `CONFIG_LWM2M_COAP_MAX_MSG_SIZE`.

Reply:
Not sure I follow, `msg->cpkt.max_len` needs to fit the entire CoAP packet (including headers),
if we set `CONFIG_LWM2M_COAP_BLOCK_SIZE` here then there's no place for headers in case of block transfer.

`CONFIG_LWM2M_COAP_MAX_MSG_SIZE` indicates the maximum size of the CoAP packet we can encode and send over the network - I don't really see why would that change for block transfer case.

Reply:
`msg->cpkt.max_len` is the size of the buffer where the CoAP packet will be encoded.

It does not tell the final size of the packet. And this has nothing to do with the block-wise option.

Above this code, the complete CoAP message is stored inside `msg->body_encode_buffer`.
Then below this, we compare if the complete message can fit this smaller buffer that is size of maximum CoAP packet that we are allowed to send. `sizeof(msg->msg_data)` is same as `CONFIG_LWM2M_COAP_MAX_MSG_SIZE`.

If we cannot fit, then we start to split the packet into smaller messages with Block-Wise.